mm2src/coins/eth.rs

```diff
@@ -939,34 +921,89 @@ pub async fn withdraw_erc721(ctx: MmArc, req: WithdrawErc721) -> WithdrawNftResu
             ))
         },
     };
-    let (gas, gas_price) = match req.fee {
-        Some(WithdrawFee::EthGas { gas_price, gas }) => {
-            let gas_price = wei_from_big_decimal(&gas_price, 9)?;
-            (gas.into(), gas_price)
-        },
-        Some(fee_policy) => {
-            let error = format!("Expected 'EthGas' fee type, found {:?}", fee_policy);
-            return MmError::err(WithdrawError::InvalidFeePolicy(error));
+    let (gas, gas_price) = get_eth_gas_details(
+        &eth_coin,
+        req.fee,
+        eth_value,
+        data.clone().into(),
+        call_addr,
+        true,
+        None,
+    )
+    .await?;
+    let _nonce_lock = eth_coin.nonce_lock.lock().await;
```

Comment:
Hmm I see, it is related what you explained in `account sequence problems` PR.
well, as some other withdraw functions in eth have it, lets also keep it for erc. just in case.

Reply:
>  as some other withdraw functions in eth have it, lets also keep it for erc. just in case

All will have the problem since they are not distributed-level locks. This lock will be by-passed when multiple mm2 instances operating same tasks concurrently

Reply:
I am not asking for fixing it in this PR, just giving an info for future implementations/optimizations